libs/cart/src/selectors/cart/cart.selector.ts

```diff
@@ -152,6 +155,17 @@ const createCartSelectors = <
 		selectCartValue,
 		cart => !cart || !cart.items || cart.items.length === 0
   );
+	const selectCartItemDiscountedRowTotal = createSelector(
+		selectCartItems,
+		(cartItems: DaffCartItem[], props) => {
+			return cartItems.reduce((acc: number, item: DaffCartItem) => {
+				if(item.item_id === props.id) {
+					return item.row_total - item.total_discount
+				}
+				return acc;
+			}, null)
```

Comment:
This shouldn't be implemented with `reduce`. I would just find the cart item in the array and do the calculation.

Reply:
> This shouldn't be implemented with reduce.
why?

Reply:
Its not a reduce operation. It doesn't depend on all the elements of the array, only one. There's no point iterating through every array element when you might find the element you need at the very beginning.

Reply:
makes sense. changed